[Bug]:

### Terraform Core Version

1.9.0

### AWS Provider Version

5.63.0

### Affected Resource(s)

* aws_ses_configuration_set
* 

### Expected Behavior

aws_ses_configuration_set resource created without error

### Actual Behavior

Request cancelled

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

### Relevant Error/Panic Output Snippet

```shell
The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-aws_v5.63.0_x5 plugin:

panic: interface conversion: interface {} is bool, not *bool

goroutine 72 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ses.resourceConfigurationSetCreate({0x17c4e188, 0xc0025d98c0}, 0xc001ad9300, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/service/ses/configuration_set.go:130 +0xf2e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc001ad9300?, {0x15897560?, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x17c4e188?, {0x17c4e188?, 0xc0025d88d0?}, 0xd?, {0x15897560?, 0xc002e18a90?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0043642a0, {0x17c4e188, 0xc0025d88d0}, 0xc0020db040, 0xc001ad9180, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0043db728, {0x17c4e188?, 0xc0025d8810?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001e92e00, {0x17c4e188?, 0xc0025d8540?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0012560a0, {0x17c4e188?, 0xc0025bfad0?}, 0xc001b14230)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1569f860, 0xc0012560a0}, {0x17c4e188, 0xc0025bfad0}, 0xc001ad8880, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000afd200, {0x17c4e188, 0xc0025bfa40}, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0, 0xc004468b70, 0x21549d98, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000afd200, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.63.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

`resource "aws_ses_configuration_set" "ses" {
  name = "default-configuration-set-${var.environment}"

  reputation_metrics_enabled = true
}`


### Steps to Reproduce

terraform apply

### Debug Output

Stack trace from the terraform-provider-aws_v5.63.0_x5 plugin:

panic: interface conversion: interface {} is bool, not *bool

goroutine 72 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ses.resourceConfigurationSetCreate({0x17c4e188, 0xc0025d98c0}, 0xc001ad9300, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/service/ses/configuration_set.go:130 +0xf2e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc001ad9300?, {0x15897560?, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x17c4e188?, {0x17c4e188?, 0xc0025d88d0?}, 0xd?, {0x15897560?, 0xc002e18a90?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0043642a0, {0x17c4e188, 0xc0025d88d0}, 0xc0020db040, 0xc001ad9180, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0043db728, {0x17c4e188?, 0xc0025d8810?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001e92e00, {0x17c4e188?, 0xc0025d8540?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0012560a0, {0x17c4e188?, 0xc0025bfad0?}, 0xc001b14230)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1569f860, 0xc0012560a0}, {0x17c4e188, 0xc0025bfad0}, 0xc001ad8880, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000afd200, {0x17c4e188, 0xc0025bfa40}, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0, 0xc004468b70, 0x21549d98, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000afd200, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.63.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Panic Output

Stack trace from the terraform-provider-aws_v5.63.0_x5 plugin:

panic: interface conversion: interface {} is bool, not *bool

goroutine 72 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ses.resourceConfigurationSetCreate({0x17c4e188, 0xc0025d98c0}, 0xc001ad9300, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/service/ses/configuration_set.go:130 +0xf2e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc001ad9300?, {0x15897560?, 0xc002e18a90})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x17c4e188?, {0x17c4e188?, 0xc0025d88d0?}, 0xd?, {0x15897560?, 0xc002e18a90?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0043642a0, {0x17c4e188, 0xc0025d88d0}, 0xc0020db040, 0xc001ad9180, {0x15897560, 0xc002e18a90})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0043db728, {0x17c4e188?, 0xc0025d8810?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001e92e00, {0x17c4e188?, 0xc0025d8540?}, 0xc002508b40)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0012560a0, {0x17c4e188?, 0xc0025bfad0?}, 0xc001b14230)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1569f860, 0xc0012560a0}, {0x17c4e188, 0xc0025bfad0}, 0xc001ad8880, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000afd200, {0x17c4e188, 0xc0025bfa40}, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0, 0xc004468b70, 0x21549d98, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000afd200, {0x17cb93a0, 0xc001832300}, 0xc0025d17a0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.63.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/38921.txt
+++ b/.changelog/38921.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ses_configuration_set: Fix crash when `reputation_metrics_enabled` is set to `true`
+```

--- a/internal/service/ses/configuration_set.go
+++ b/internal/service/ses/configuration_set.go
@@ -127,7 +127,7 @@ func resourceConfigurationSetCreate(ctx context.Context, d *schema.ResourceData,
 	if v := d.Get("reputation_metrics_enabled"); v.(bool) {
 		input := &ses.UpdateConfigurationSetReputationMetricsEnabledInput{
 			ConfigurationSetName: aws.String(configurationSetName),
-			Enabled:              aws.ToBool(v.(*bool)),
+			Enabled:              v.(bool),
 		}
 
 		_, err := conn.UpdateConfigurationSetReputationMetricsEnabled(ctx, input)

